### ISSUE TITLE
Constructors of controllers instantiated via storyboards aren't called

### ISSUE THREAD

**Issue by badlogic** [robovm/robovm#894]
Given the following `UITableViewController`:

```java
@CustomClass("TestController")
public class TestController extends UITableViewController {
    private final List<String> items = new ArrayList<String>();

    @Override
    public void viewDidLoad() {
        super.viewDidLoad();
        items.add("Cat");
        items.add("Dog");
        items.add("Bird");
    }
...
```

I write this up to a storyboard. When the app loads, the `TestController` is unmarshalled and the `viewDidLoad` method is called. The app crashes with an NPE on the first call to `items#add`.

**Comment by badlogic** [comment]
`viewDidLoad` is called before the default constructor and instance initializer code is called :(

**Comment by BlueRiverInteractive** [comment]
Possibly related: You cannot override the constructor of a native class like UIView. You can, but when objc creates a new instance of your subclass, it will not invoke the Java constructor. So if you want to execute code on object creation, you have to override one of the protected initXXX methods.

**Comment by ntherning** [comment]
This is "by design". :-) If the ObjC side instantiates an instance of your custom class no constructor will be called on the Java side. You can set up of fields in `afterMarshaled()` or in `viewDidLoad()`. Yeah, this sucks and we should do something about it but it's not trivial.

**Comment by badlogic** [comment]
What's the problem with calling the (default) constructor?

**Comment by ntherning** [comment]
It ultimately calls `ObjCObject()` which calls `alloc` and `init` on the ObjC side. We need a way to know in the default constructor that it shouldn't do `alloc`/`init`. Also, we don't do anything ATM when the object is allocated/inited on the ObjC side. It's not until it is passed over to the Java side we create an instance on that side. We need a way to hook into the allocation/initialization on the ObjC side.

**Comment by ntherning** [comment]
We definitely need to fix this and make it possible to bind constructors. But there's not enough time for 1.1. Will postpone until 1.2.

**Comment by simophin** [comment]
This sort of breaks Java language, when can this be fixed?

**Comment by JohnColanduoni** [comment]
As a possible suggestion for how to fix this, you might want to take a look at how [Xamarin](http://developer.xamarin.com/api/type/MonoTouch.Foundation.NSObject/) does it. Instead of having an init method, constructors for both the "create from managed code" (parameterless) and "create managed wrapper for unmanaged object" (with a pointer parameter) cases are provided.

The main disadvantage to this is that you cannot make managed versions of init return nil, but I suspect this an uncommon use case.
